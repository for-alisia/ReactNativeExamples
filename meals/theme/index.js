export default {
  colors: {
    primary: '#4a148c',
    accentColor: '#ff6f00',
    light: 'white'
  },
  randomColors: [
    '#f5428d',
    '#f54242',
    '#f5a442',
    '#f5d142',
    '#368dff',
    '#41d95d',
    '#46BFC7',
    '#439027',
    '#AD4CAD',
    '#47fced',
  ],
};
