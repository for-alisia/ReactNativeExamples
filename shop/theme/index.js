const theme = {
  colors: {
    primary: '#DC8080',
    primaryLight: '#ECBDBC',
    secondary: '#D8C496',
    secondaryLight: '#F8ECC9',
    accent: '#CA5454',
    accentDark: '#6B5344',
    light: '#F6F6F6',
    white: '#ffffff',
    grey: '#685E59',
    greyLight: '#ADA19B',
    dark: '#3C3735',
    yellow: '#FF9921',
  },

  fonts: {
    montserratReg: 'montserrat-regular',
    monserratBold: 'montserrat-bold',
    playfair: 'playfair',
    playfairBold: 'playfair-bold',
  },

  fontSize: {
    xs: 12,
    s: 16,
    m: 24,
    l: 32,
    xl: 102,
  },

  padding: {
    xs: 4,
    s: 8,
    m: 12,
    l: 16,
    xl: 24,
    xxl: 32,
  },

  margin: {
    xs: 4,
    s: 8,
    m: 12,
    l: 16,
    xl: 24,
    xxl: 32,
  },

  borderRadius: {
    xs: 4,
    s: 8,
    m: 16,
    l: 24,
  },

  activeOpacity: 0.75,
};

export default theme;
