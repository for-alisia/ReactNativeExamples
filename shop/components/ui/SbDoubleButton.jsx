import React from 'react';
import { StyleSheet, View } from 'react-native';

// Components
import SbText from './SbText';
import SbTouchable from './SbTouchable';

// Theme
import theme from '../../theme';

// Supportes types: 'solid' and 'outline'

const SbDoubleButton = (props) => {
  const {
    onRightPress,
    onLeftPress,
    type,
    style,
    disabledRight,
    disabledLeft,
    leftLabel,
    rightLabel,
  } = props;

  const typeStyles = type || 'solid';
  const textStyles = typeStyles === 'solid' ? 'textSolid' : 'textOutlined';

  return (
    <View style={styles.wrapper}>
      <SbTouchable onPress={!disabledLeft ? onLeftPress : () => {}}>
        <View
          style={{
            ...styles.container,
            ...styles.containerLeft,
            ...styles[typeStyles],
            ...style,
            opacity: disabledLeft ? 0.68 : 1,
          }}
        >
          <SbText style={{ ...styles.text, ...styles[textStyles] }}>{leftLabel}</SbText>
        </View>
      </SbTouchable>
      <SbTouchable onPress={!disabledRight ? onRightPress : () => {}}>
        <View
          style={{
            ...styles.container,
            ...styles.containerRight,
            ...styles[typeStyles],
            ...style,
            opacity: disabledRight ? 0.68 : 1,
          }}
        >
          <SbText style={{ ...styles.text, ...styles[textStyles] }}>{rightLabel}</SbText>
        </View>
      </SbTouchable>
    </View>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
  },
  container: {
    paddingVertical: theme.padding.s,
    paddingHorizontal: theme.padding.xl,
    borderRadius: theme.borderRadius.l,
    alignSelf: 'center',
  },
  containerLeft: {
    borderTopRightRadius: 0,
    borderBottomRightRadius: 0,
    borderRightWidth: 0,
  },
  containerRight: {
    borderTopLeftRadius: 0,
    borderBottomLeftRadius: 0,
  },
  solid: {
    backgroundColor: theme.colors.primary,
  },
  outline: {
    backgroundColor: 'transparent',
    borderColor: theme.colors.primary,
    borderWidth: 1,
  },
  text: {
    fontSize: theme.fontSize.s,
    textAlign: 'center',
  },
  textSolid: {
    color: theme.colors.light,
  },
  textOutlined: {
    color: theme.colors.primary,
  },
});

export default SbDoubleButton;
