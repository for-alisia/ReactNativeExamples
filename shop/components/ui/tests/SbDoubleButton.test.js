// @ts-nocheck
// Dependencies
import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { toHaveStyle } from '@testing-library/jest-native';

// Component to test
import SbDoubleButton from '../SbDoubleButton';

// Theme
import theme from '../../../theme';

describe('SbDoubleButton UI Component', () => {
  // To use toHaveStyle
  expect.extend({ toHaveStyle });
  const textLeft = 'Hello';
  const textRight = 'World';

  it('renders button with a correct text', () => {
    const { getByText, toJSON } = render(
      <SbDoubleButton leftLabel={textLeft} rightLabel={textRight} />
    );
    const foundLeftText = getByText(textLeft);
    const foundRightText = getByText(textRight);

    expect(foundLeftText).toBeTruthy();
    expect(foundRightText).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls handlers functions correctly', () => {
    const fnRight = jest.fn();
    const fnLeft = jest.fn();

    const { getByText } = render(
      <SbDoubleButton
        leftLabel={textLeft}
        rightLabel={textRight}
        onRightPress={fnRight}
        onLeftPress={fnLeft}
      />
    );

    const left = getByText(textLeft);
    const right = getByText(textRight);

    fireEvent.press(left);
    expect(fnLeft).toHaveBeenCalled();

    fireEvent.press(right);
    expect(fnRight).toHaveBeenCalled();
  });

  it('works correctly with disabled buttons', () => {
    const fnRight = jest.fn();
    const fnLeft = jest.fn();

    const { getByText } = render(
      <SbDoubleButton
        leftLabel={textLeft}
        rightLabel={textRight}
        onRightPress={fnRight}
        onLeftPress={fnLeft}
        disabledRight={true}
        disabledLeft={true}
      />
    );

    const left = getByText(textLeft);
    const right = getByText(textRight);

    fireEvent.press(left);
    expect(fnLeft).not.toHaveBeenCalled();

    fireEvent.press(right);
    expect(fnRight).not.toHaveBeenCalled();
  });

  it('renders outline styled button', () => {
    const { getByText } = render(
      <SbDoubleButton leftLabel={textLeft} rightLabel={textRight} type="outline" />
    );

    const left = getByText(textLeft);
    const right = getByText(textRight);

    expect(left).toHaveStyle({
      color: theme.colors.primary,
    });

    expect(right).toHaveStyle({
      color: theme.colors.primary,
    });
  });
});
