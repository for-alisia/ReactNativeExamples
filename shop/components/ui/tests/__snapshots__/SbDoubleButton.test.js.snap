// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SbDoubleButton UI Component renders button with a correct text 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#DC8080",
          "borderBottomRightRadius": 0,
          "borderRadius": 24,
          "borderRightWidth": 0,
          "borderTopRightRadius": 0,
          "opacity": 1,
          "paddingHorizontal": 24,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#F6F6F6",
            "fontFamily": "montserrat-regular",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Hello
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#DC8080",
          "borderBottomLeftRadius": 0,
          "borderRadius": 24,
          "borderTopLeftRadius": 0,
          "opacity": 1,
          "paddingHorizontal": 24,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#F6F6F6",
            "fontFamily": "montserrat-regular",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        World
      </Text>
    </View>
  </View>
</View>
`;
